wolfcrypt/src/port/silabs/silabs_ecc.c

```diff
@@ -0,0 +1,319 @@
+
+    sl_stat = sl_se_get_storage_size(&key->key, &key->key.storage.location.buffer.size);
+    key->key.storage.location.buffer.pointer = key->key_raw;
+
+    sl_stat = sl_se_generate_key(&(key->cmd_ctx),
+                                 &(key->key));
+
+    key->type = ECC_PRIVATEKEY;
+
+    /* copy key to mp components */
+    mp_read_unsigned_bin (key->pubkey.x,
+                          key->key.storage.location.buffer.pointer,
+                          keysize);
+    mp_read_unsigned_bin (key->pubkey.y,
+                          key->key.storage.location.buffer.pointer  + keysize,
+                          keysize);
+    mp_read_unsigned_bin (&key->k,
+                          key->key.storage.location.buffer.pointer + 2 * keysize,
+                          keysize);
+
+    return (sl_stat == SL_STATUS_OK) ? 0 : WC_HW_E;
+}
+
+int silabs_ecc_import(ecc_key* key, word32 keysize)
+{
+    sl_status_t sl_stat;
+    int err = MP_OKAY;
+    word32 used = keysize;
+
+    key->key.type = silabs_map_key_type(key->dp->id);
+    key->key.size = keysize;
+    key->key.storage.method = SL_SE_KEY_STORAGE_EXTERNAL_PLAINTEXT;
+    key->key.flags = SL_SE_KEY_FLAG_ASYMMETRIC_BUFFER_HAS_PRIVATE_KEY
+        | SL_SE_KEY_FLAG_ASYMMETRIC_BUFFER_HAS_PUBLIC_KEY
+        | SL_SE_KEY_FLAG_ASYMMMETRIC_SIGNING_ONLY
+        ;
+
+    sl_stat = sl_se_get_storage_size(&key->key, &key->key.storage.location.buffer.size);
+    key->key.storage.location.buffer.pointer = key->key_raw;
+    if (sl_stat != SL_STATUS_OK)
+      return WC_HW_E;
+
+    key->type = ECC_PRIVATEKEY;
+
+    /* copy key from mp components */
+    if (err == MP_OKAY)
+        err = wc_export_int(key->pubkey.x, key->key.storage.location.buffer.pointer,
+                            &used, keysize,
+                            WC_TYPE_UNSIGNED_BIN);
+    if (err == MP_OKAY)
+        err = wc_export_int(key->pubkey.y, key->key.storage.location.buffer.pointer + keysize,
+                            &used, keysize,
+                            WC_TYPE_UNSIGNED_BIN);
+    if (err == MP_OKAY)
+        err = wc_export_int(&key->k, key->key.storage.location.buffer.pointer + 2 * keysize,
+                            &used, keysize,
+                            WC_TYPE_UNSIGNED_BIN);
+
+    return err;
+}
+
+int silabs_ecc_import_private(ecc_key* key, word32 keysize)
+{
+    sl_status_t sl_stat;
+    word32 keySz = keysize;
+    key->key.type = silabs_map_key_type(key->dp->id);
+    key->key.size = key->dp->size;
+    key->key.storage.method = SL_SE_KEY_STORAGE_EXTERNAL_PLAINTEXT;
+    key->key.flags = SL_SE_KEY_FLAG_ASYMMETRIC_BUFFER_HAS_PRIVATE_KEY
+        | SL_SE_KEY_FLAG_ASYMMMETRIC_SIGNING_ONLY
+        ;
+
+    sl_stat = sl_se_get_storage_size(&key->key, &key->key.storage.location.buffer.size);
+    key->key.storage.location.buffer.pointer = key->key_raw;
+    if (sl_stat != SL_STATUS_OK)
+          return WC_HW_E;
+
+    return wc_export_int(&key->k, key->key.storage.location.buffer.pointer,
```

Comment:
This passes in a pointer to `keySz`, but its return value is lost / not used. Perhaps there is an improvement that can be made?

Reply:
Still want to look at this more to see how to best use or check this.

Reply:
I've added an extra check here. I could have called `mp_to_unsigned_bin` directly, but I like reusing the code that checks the buffer size.